docs/setup.md

```diff
@@ -0,0 +1,36 @@
+# ACK `test-infra` dev env setup
+
+This document will walk you through setting up your ACK test-infra development
+environment.
+
+## Pre-requisites
+
+We will use `pipx` to install some Python3 CLI tools that make it easy for us
+to do isolated development on our local machines without requiring root
+privileges or installing software outside the local user context.
+
+Install and configure `pipx` like so:
+
+```bash
+python3 -m pip install --user pipx
+python3 -m pipx ensurepath
+```
+
+Install the `pipenv` tool for creating Python3 virtual environments:
+
+```bash
+pipx install pipenv
```

Comment:
Using `pipx` to install `pipenv`? I've never heard about this before

Reply:
The benefits of pipx I believe will become super evident as we end up having a common test-infra Python framework and CLI application along with e2e tests and potentially Python CLI applications in lots of the service controller repositories. Having pipx's ease of dealing with Python code that is arranged in local Git repositories will be invaluable I think.